### PR TITLE
Routing fixes extended

### PR DESCRIPTION
**Pull request by siemiatj**
This fixes some issues with tokens as well as removes some unnecessary re-renders.

Related to #11015

### CODE REVIEW

**Comment by siemiatj** [comment]
e2e https://dashboard.cypress.io/projects/5yp4q1/runs/1238/overview

File: frontend/src/routes/TokenRoute.js [hunk 1]
```diff
@@ -15,14 +16,13 @@ import history from '../services/History';
...
+  const state = useStore().getState();
 
-  auth
-    .tokenLogin(tokenId)
-    .then(() => history.push('/'))
-    .catch(() => {
-      auth.clearRedirectRoute();
-      history.push('/login');
+  if (!auth.authRequestPending() && !state.appHandler.isLogged) {
+    auth.tokenLogin(tokenId).then(() => {
```
Comment (petrican) [inline frontend/src/routes/TokenRoute.js:22]: auth.tokenLogin(tokenId).then(() => history.push('/'));

**Review by petrican — APPROVED** [review]
Some minor cosmetic changes to the code requested. Other than that lgtm. Thanks

**Comment by petrican** [comment]
Also I believe we should merge this on Monday if possible.

File: frontend/src/components/notifications/NotificationHandler.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import PropTypes from 'prop-types';
-import React, { Component } from 'react';
+import React, { PureComponent } from 'react';
 import { connect } from 'react-redux';
 
 import Notification from './Notification';
 
-class NotificationHandler extends Component {
+const EMPTY_OBJECT = { notifications: {} };
```
[the remark this answers is not in the document]
Comment (siemiatj) [reply frontend/src/components/notifications/NotificationHandler.js:7]: We can't as other components will import the same object, and can modify it which then could lead to unexpected errors. We've been through that before :)

**Comment by siemiatj** [comment]
e2e https://jenkins.metasfresh.com/job/ops/job/run_e2e_tests/1434/

**Pull request merged**